### PR TITLE
Added Notifications tab to BuddyPress integration.

### LINKED ISSUE #627: BuddyPress profile doesn't have a LifterLMS "Notifications" tab
_62167_lifter_lms_notifications_to_be_listed_in_buddypress_area_-_science_cosmos

### PR DESCRIPTION
**Pull request by yojance**
## Description
Added Notifications tab to BuddyPress integration.

Fixes #627 

## How has this been tested?
Code has been tested locally and it works as expected. It's using the same template from `myaccount/my-notifications.php`

## Screenshots
lifterlms-buddypress-notifications-tab

## Todo
The pagination will not keep the user in the same page. Clicking on Next will take them to `my-courses/notifications/?sdpage=2` instead of `/members/yojance/courses/notifications/?sdpage=2`. I'm open to suggestions if you'd like this behavior changed.

## Checklist:
- [x] My code has been tested.

### CODE REVIEW

**Comment by thomasplevy** [comment]
@yojance looks pretty solid but we can probably prevent a disjointed user experience related to your TODO items with the following updates:

+ Add an `apply_filter` on the `$url` variable in the `LLMS_Student_Dashboard::output_notifications_content()` function
+ Use an `add_filter` to modify the default URL in the BP function added to this PR and the `remove_filter` immediately after output

This shouldn't be too much additional code or overhead and will prevent users from being bumped out of their BP profile.

Do you want to take a stab at that an update the PR?

Thanks for the effort thus far regardless

**Comment by thomasplevy** [comment]
Travis fail is acceptable. Timestamp delta issue on PHP 7.0 only.

**Comment by thomasplevy** [comment]
Woops, didn't mean to close

File: includes/integrations/class.llms.integration.buddypress.php [hunk 1]
```diff
@@ -96,6 +96,15 @@ public function add_profile_nav_items() {
 			'screen_function' => array( $this,'certificates_screen' ),
 			'user_has_access' => $is_my_profile,
 		));
+
+		bp_core_new_subnav_item(array(
+			'name'            => __( 'Notifications', 'lifterlms' ),
+			'slug'            => 'notifications',
+			'parent_slug'     => 'courses',
+			'parent_url'      => $parent_url,
+			'screen_function' => array( $this,'notifications_screen' ),
```
Comment (thomasplevy) [inline includes/integrations/class.llms.integration.buddypress.php]: code climate identified a code styling error here, there should be a space between  the comma and single quote on this line (105)

**Comment by yojance** [comment]
@thomasplevy 

I'm not 100% sure on how to implement your suggestions. I've taken it as far as I could but I can't seem to get it to work. I believe I've seen the technique you are suggesting but I just can't get it to work. Help!

**Comment by thomasplevy** [comment]
@yojance your code should be working but I'm not certain why it's not. I'll take a look at it and see what I can find

**Pull request closed without merging**